tests/test_common_view_logic.py

```diff
@@ -1,389 +1,182 @@
-        expected_context = {
-            "current_page": 1,
-            "total_pages": 2,
-            "featured_articles": [],
-            "articles": [
-                {
-                    "author": "test_author",
-                    "categories": [1, 2],
-                    "display_category": "test_category",
-                    "featured_media": "test",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test"],
-                },
-                {
-                    "author": "test_author",
-                    "categories": [2, 3],
-                    "display_category": "test_category",
-                    "featured_media": "test2",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test2"],
-                },
-            ],
-            "groups": {1: "test_group"},
-            "upcoming": [],
-            "used_categories": {
-                1: "test_category",
-                2: "test_category",
-                3: "test_category",
-            },
-        }
-        self.maxDiff = None
-        self.assertEqual(context, expected_context)
-
-    @patch("canonicalwebteam.blog.wordpress_api.get_group_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_category_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_user")
-    @patch("canonicalwebteam.blog.wordpress_api.get_media")
-    def test_building_index_context_with_featured_articles(
-        self, get_media, get_user, get_category_by_id, get_group_by_id
-    ):
-        get_media.return_value = "test_image"
-        get_user.return_value = "test_author"
-        get_category_by_id.return_value = "test_category"
-        get_group_by_id.return_value = "test_group"
-        articles = [
-            {
-                "featured_media": "test",
-                "author": "test",
-                "categories": [1, 2],
-                "group": [1],
-                "tags": ["test"],
-            },
-            {
-                "featured_media": "test2",
-                "author": "test2",
-                "categories": [2, 3],
-                "group": [1],
-                "tags": ["test2"],
-            },
+    @vcr.use_cassette("fixtures/vcr_cassettes/get_complete_article.yaml")
+    def test_get_complete_article(self):
+        article = api.get_article(
+            "canonical-announces-support-for"
+            "-ubuntu-on-windows-subsystem-for-linux-2"
+        )
+        article["categories"] = [1453]
+
+        complete_article = get_complete_article(article)
+        self.assertEqual(complete_article["author"]["name"], "Canonical")
+        self.assertEqual(
+            complete_article["image"]["id"], article["featured_media"]
+        )
+        self.assertEqual(complete_article["date"], "6 May 2019")
+        # Test for length of 339, and if > than 340, [...] is added
```

Comment:
this is why I don't like comments. Sometimes they lie! :P
Where do you test this?
 
`self.assertLessEqual(len(complete_article["excerpt"]["raw"]), 344)` this just tells me that the raw is 344 characters.

What you want:

`self.assertEqual(complete_article["excerpt"]["raw"][-5:], "[...]")`

Reply:
Hm, but what I want to test is that the amount of characters here is always less than 344, which is the theoretical maximum.

Reply:
I agree with what you say but the comment tells me something else.

Reply:
Unfortunately I cant test the "[...]" here since the article excerpt might be less than 340, in which case this is not added.

Rather than putting in an if here I think this calls for  a refactor of some sort, that encapsulates the trimming logic and can be tested in isolation.